mcs/class/System/Test/System.Net.Sockets/SocketAcceptAsyncTest.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Threading;
+using System.Net;
+using System.Net.Sockets;
+using NUnit.Framework;
+
+namespace MonoTests.System.Net.Sockets
+{
+	[TestFixture]
+	public class SocketAcceptAsyncTest
+	{
+		private Socket _listenSocket;
+		private Socket _clientSocket;
+		private Socket _serverSocket;
+		private Socket _acceptedSocket;
+		private ManualResetEvent _readyEvent;
+		private ManualResetEvent _mainEvent;
+
+		[TestFixtureSetUp]
+		public void SetUp()
+		{
+			_readyEvent = new ManualResetEvent(false);
+			_mainEvent = new ManualResetEvent(false);
+
+			ThreadPool.QueueUserWorkItem(_ => StartListen());
```

Comment:
Please inline this StartListen method as a delegate so that we do not use global variables.